salt/states/zabbix_host.py

```diff
@@ -151,11 +151,18 @@ def _interface_format(interfaces_data):
                     details_community = val_details.get(
                         "community", "{$SNMP_COMMUNITY}"
                     )
-                details = {
-                    "version": details_version,
-                    "bulk": details_bulk,
-                    "community": details_community,
-                }
+                # a community string should only be provided in legacy versions of SNMP
+                if details_version == '3':
+                    details = { 
+                            "version": details_version,
+                            "bulk": details_bulk,
+                    }
+                else:
+                    details = { 
+                            "version": details_version,
+                            "bulk": details_bulk,
+                            "community": details_community,
+                    }
```

Comment:
Instead creating this second test, why not just remove the line 157 and add an `else` section to the following `if` and set the `community` there?